index.js

```diff
@@ -25,7 +25,8 @@ module.exports = render;
 
 function render(view, opts) {
   opts = opts || {};
-  var ext = opts.engine || extname(view).slice(1);
+  var _extname = extname(view).slice(1);
+  var ext = opts.engine || _extname === 'jade' ? 'pug' : _extname;
```

Comment:
co-render's logic depend on template file's extname , so if co-render doesn't process it, 

```
-  var ext = opts.engine || extname(view).slice(1);
```

this code will be wrong